include/zenoh/api/session.hxx

```diff
@@ -433,14 +433,14 @@ class Session : public Owned<::z_owned_session_t> {
                                                       SubscriberOptions&& options = SubscriberOptions::create_default(),
                                                       ZResult* err = nullptr) const {
         static_assert(
-            std::is_invocable_r<void, C, const Sample&>::value,
```

Comment:
const was correct here, since if function is invocable with const Sample&, it is also automatically invocable with Sample&